opensrp-giz-malawi/src/main/java/org/smartregister/giz/viewholder/VillageDoseViewHolder.java

```diff
@@ -0,0 +1,61 @@
+package org.smartregister.giz.viewholder;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+
+import org.smartregister.giz.R;
+import org.smartregister.giz.domain.VillageDose;
+import org.smartregister.giz.util.GizUtils;
+import org.smartregister.view.ListContract;
+import org.smartregister.view.viewholder.ListableViewHolder;
+
+import java.util.Map;
+
+public class VillageDoseViewHolder extends ListableViewHolder<VillageDose> {
+
+    private View currentView;
+    private TextView tvName;
+    private LinearLayout linearLayout;
+    private Context context;
+
+    public VillageDoseViewHolder(@NonNull View itemView, Context context) {
+        super(itemView);
+        this.currentView = itemView;
+        tvName = itemView.findViewById(R.id.tvName);
+        linearLayout = itemView.findViewById(R.id.linearLayout);
+        this.context = context;
+    }
+
+    @Override
+    public void bindView(VillageDose villageDose, ListContract.View<VillageDose> view) {
+        tvName.setText(villageDose.getVillageName());
+
+        for (Map.Entry<String, Integer> entry : villageDose.getRecurringServices().entrySet()) {
```

Comment:
Check for null `villageDose.getRecurringServices()`

Reply:
Done